### PR TITLE
Limit time for turn

### LINKED ISSUE #85: Time limit per card
To avoid having long discussions for one single card, one can add a (configurable) timebox per card.

When dealing a card a countdown is shown. After this either the game should move on, or (as a radical alternative) all players auto-pass.

It should be configurable, how long and whether there is a timebox.

### PR DESCRIPTION
**Pull request by matthewejones**
Continued on from #85

- Made timer configurable including custom time
- Added some tests
- Timer now disappears once everyone's passed
- Changed colour and font:
  - I just changed these to use the reactstrap colours and the font for the rest of the page. 
  - I removed the "Times Up..." message because I though it looked better just showing 00:00 but we can change this quite easily. Another alternative could be to just make it pass for you but this might be really annoying if you're half way through adding a threat
- fixed bug where if a player played their card when the tab is in the background, the timer loaded with the correct remaining time but didn't start counting down until the tab was opened. This meant that the timer showed the wrong value.

Still to do maybe:
- make custom duration accept mm:ss instead of just seconds?
- Refactor the form input into its own component (it's become rather a lot of code)

Resolves #85

### CODE REVIEW

**Comment by ChristophNiehoff** [comment]
@matthewejones Pretty cool!

Would it be possible to change to color gradient of the timer? It would be cool if the timer was green in the first half, then orange in the third quarter and red in the last quarter.

  Is there some way to embed the timer in a optically nicer way? Maybe put it into a box with white background?  Maybe another location (e.g. bottom-left right over the cards?  Would it fit there?)  What do you think?

**Comment by matthewejones** [comment]
> @matthewejones Pretty cool!
> 
> Would it be possible to change to color gradient of the timer? It would be cool if the timer was green in the first half, then orange in the third quarter and red in the last quarter.
> 
> Is there some way to embed the timer in a optically nicer way? Maybe put it into a box with white background? Maybe another location (e.g. bottom-left right over the cards? Would it fit there?) What do you think?

yeah, changing the durations of the colours is really easy.

Yeah that's something I was thinking about but wasn't really sure where the best place to put it is. I'll try some different places, see where it fits best

**Comment by matthewejones** [comment]
Let me know which place you like most out of these or if you have any other suggestions. I've made the middle solid white but this could be anything or transparent. I think I prefer bottom left or bottom right over the cards. This way it doesn't get in the way of adding threats to the model. 

image

image

image

**Comment by ChristophNiehoff** [comment]
I also like bottom left. Am I correct thatothe timer is not visible while drawing a card?

**Comment by matthewejones** [comment]
> I also like bottom left. Am I correct thatothe timer is not visible while drawing a card?

Yeah, it's only there when adding threats, once everyone's passed it disappears

File: src/client/components/sidebar/sidebar.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { faDownload } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { API_PORT } from '../../../utils/constants';
 import Footer from '../footer/footer';
+import Timer from '../timer/timer';
```
Comment (ChristophNiehoff) [inline src/client/components/sidebar/sidebar.js:12]: This import statement can be removed.

File: src/client/components/timer/timer.css [hunk 2]
```diff
@@ -0,0 +1,28 @@
+.timer-wrapper {
+  display: flex;
+  justify-content: center;
+  border-radius: 50%;
+  margin: 20px;
+}
+
+.timer {
+  /* font-family: "Montserrat"; */
```
Comment (ChristophNiehoff) [inline src/client/components/timer/timer.css:9]: Can this comment be removed?

File: src/client/pages/create.js [hunk 3]
```diff
@@ -220,6 +240,37 @@ class Create extends React.Component {
               </Col>
             </FormGroup>
             <hr />
+            <FormGroup row>
+              <Label for="turnDuration" sm={2}>Turn Duration</Label>
+              <Col sm={10}>
+                <Input type="select" name="turnDuration" id="turnDuration" onChange={e => this.onTurnDurationUpdated(e)} value={this.state.turnDuration}>
+                  <option value={0}>No Timer</option>
```
Comment (ChristophNiehoff) [inline src/client/pages/create.js:247]: I think we should keep it simple. We only need the values
* No Timer
* 3min
* 5min (default)
* 10min

And I would remove the `custom` logic.

**Review by ChristophNiehoff — CHANGES_REQUESTED** [review]

File: src/client/pages/create.js (hunk 3 above)
Comment (matthewejones) [reply src/client/pages/create.js:247]: Oh yeah, I did mean to remove custom. Just forgot

**Review by ChristophNiehoff — APPROVED** [review]

**Pull request merged**